In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv("NFCS 2018 State Data 190603.csv")
df

NFCSID  STATEQ  CENSUSDIV  CENSUSREG  A3  A3Ar_w  A3B  A4A_new_w  \
0      2018010001      48          9          4   2       5   11          1   
1      2018010002      10          5          3   2       2    8          1   
2      2018010003      44          7          3   2       2    8          1   
3      2018010004      10          5          3   2       1    7          1   
4      2018010005      13          8          4   1       2    2          1   
...           ...     ...        ...        ...  ..     ...  ...        ...   
27086  2018037087      51          8          4   2       3    9          1   
27087  2018037088      20          1          1   2       2    8          2   
27088  2018037089      20          1          1   1       2    2          2   
27089  2018037090      20          1          1   2       1    7          2   
27090  2018037091      20          1          1   2       1    7          2   

       A5_2015  A6  ...  M42  M6  M7  M8  M31  M9 M10    wgt_n2    wgt_d2  \
0            5   4  ...        1   3  98   98  98   1  0.683683  0.519642   
1            6   1  ...        1   3  98    3   1  98  0.808358  2.516841   
2            6   1  ...        1   1  98   98   1  98  1.021551  1.896192   
3            6   2  ...    7  98  98   4    4   2  98  0.808358  2.516841   
4            6   1  ...        1   3  98    2   1  98  0.448075  0.614733   
...        ...  ..  ...  ...  ..  ..  ..  ...  ..  ..       ...       ...   
27086        2   1  ...        1   2   1    1   1   2  0.404002  0.183442   
27087        2   1  ...        1   2  98   98  98  98  0.475130  0.737766   
27088        6   2  ...        1   3   4    3  98  98  0.531368  0.713412   
27089        6   2  ...        1   2   3    3   1   2  0.377318  0.546359   
27090        4   2  ...    5   1   3   1    2   1   1  0.475130  0.737766   

         wgt_s3  
0      1.095189  
1      0.922693  
2      0.671093  
3      0.922693  
4      1.232221  
...         ...  
27086  0.808332  
27087  1.116158  
27088  2.392632  
27089  0.965486  
27090  1.116158  

[27091 rows x 128 columns]

In [3]:
df.isnull().values.any()

False

# Grabbing only relevant columns

In [4]:
my_list = df.columns.values.tolist()
#my_list

We might want to consider the following points
* J4
* J33_40
* J42_1
* B2
* C40
* F2_6
* G20
* all of G30
* G22
* G25_2
* H1
* H30_3
* M4

In [5]:
skinny_df = df[['NFCSID',
 'STATEQ',
 'CENSUSDIV',
 'CENSUSREG',
 'A3',
 'A3Ar_w',
 'A3B',
 'A4A_new_w',
 'A5_2015',
 'A6',
 'A7',
 'A7A',
 'A11',
 'A8',
 'AM21',
 'AM30',
 'AM31',
 'AM22',
 'X3',
 'X4',
 'A9',
 'A40',
 'A10',
 'A10A',
 'A21_2015',
 'A22_2015',
 'A14',
 'A41',
 'J1',
 'J2',
 'J3',
 'J5',
 'J20',
 'J32',
 'J41_1',
 'J41_2',
 'C1_2012',
 'C4_2012',
 'C5_2012',
 'D40',
 'EA_1',
 'E7',
 'E8',
 'E20',
 'E15_2015',
 'F2_1',
 'F2_2',
 'F2_3',
 'F2_4',
 'F2_5',
 'G38',
 'M40',
 'M20',
 'M21_1',
 'M21_2_2015',
 'M21_3',
 'M21_4',
 'M41',
 'M42',
 'M6',
 'M9',
 'M10',
 'wgt_n2',
 'wgt_d2',
 'wgt_s3']]
skinny_df

NFCSID  STATEQ  CENSUSDIV  CENSUSREG  A3  A3Ar_w  A3B  A4A_new_w  \
0      2018010001      48          9          4   2       5   11          1   
1      2018010002      10          5          3   2       2    8          1   
2      2018010003      44          7          3   2       2    8          1   
3      2018010004      10          5          3   2       1    7          1   
4      2018010005      13          8          4   1       2    2          1   
...           ...     ...        ...        ...  ..     ...  ...        ...   
27086  2018037087      51          8          4   2       3    9          1   
27087  2018037088      20          1          1   2       2    8          2   
27088  2018037089      20          1          1   1       2    2          2   
27089  2018037090      20          1          1   2       1    7          2   
27090  2018037091      20          1          1   2       1    7          2   

       A5_2015  A6  ...  M21_3  M21_4  M41  M42  M6  M9 M10    wgt_n2  \
0            5   4  ...                           1  98   1  0.683683   
1            6   1  ...                           1   1  98  0.808358   
2            6   1  ...                           1   1  98  1.021551   
3            6   2  ...      1           3    7  98   2  98  0.808358   
4            6   1  ...                           1   1  98  0.448075   
...        ...  ..  ...    ...    ...  ...  ...  ..  ..  ..       ...   
27086        2   1  ...                           1   1   2  0.404002   
27087        2   1  ...                           1  98  98  0.475130   
27088        6   2  ...                           1  98  98  0.531368   
27089        6   2  ...                           1   1   2  0.377318   
27090        4   2  ...      1           1    5   1   1   1  0.475130   

         wgt_d2    wgt_s3  
0      0.519642  1.095189  
1      2.516841  0.922693  
2      1.896192  0.671093  
3      2.516841  0.922693  
4      0.614733  1.232221  
...         ...       ...  
27086  0.183442  0.808332  
27087  0.737766  1.116158  
27088  0.713412  2.392632  
27089  0.546359  0.965486  
27090  0.737766  1.116158  

[27091 rows x 65 columns]

# Looking for unanswered questions

### Looking for how many there are in total

In [6]:
search99= skinny_df.apply(pd.value_counts)
#search99

In [7]:
search99 = search99.loc['99']

In [8]:
search99.sum()

1312.0

#### That's a lot of unanswered quetions. How many individuals didn't answer a question?

In [9]:
grab99 = skinny_df[(skinny_df == '99').any(axis=1)]

In [10]:
grab99

NFCSID  STATEQ  CENSUSDIV  CENSUSREG  A3  A3Ar_w  A3B  A4A_new_w  \
34     2018010035      48          9          4   1       6    6          1   
36     2018010037      33          2          1   2       6   12          1   
40     2018010041      41          5          3   2       3    9          1   
72     2018010073      51          8          4   1       6    6          1   
149    2018010150      48          9          4   1       3    3          1   
...           ...     ...        ...        ...  ..     ...  ...        ...   
26857  2018036858      12          9          4   1       2    2          2   
26954  2018036955      12          9          4   2       4   10          2   
27006  2018037007       8          5          3   1       3    3          1   
27024  2018037025       8          5          3   1       6    6          1   
27072  2018037073      51          8          4   1       2    2          2   

       A5_2015  A6  ...  M21_3  M21_4  M41  M42  M6  M9 M10    wgt_n2  \
34           4   1  ...                           1  98  98  0.707918   
36           2   1  ...                           1   1  98  2.219286   
40           2   1  ...                          98  98  98  1.026420   
72           2   1  ...                          99   1  98  0.485971   
149          2   2  ...                           3   1   2  0.726409   
...        ...  ..  ...    ...    ...  ...  ...  ..  ..  ..       ...   
26857        6   1  ...                          99   1   2  1.997203   
26954        4   2  ...                           1   1   2  1.800757   
27006        6   1  ...      1           3    7  99   1   1  1.006108   
27024        6   1  ...     99     99    2    5   3   1   2  0.980497   
27072        4   1  ...                           1   1   1  0.708752   

         wgt_d2    wgt_s3  
34     0.514290  1.096894  
36     1.532573  1.096985  
40     0.654446  0.872951  
72     0.225086  1.427754  
149    0.503368  0.949739  
...         ...       ...  
26857  0.182560  1.590288  
26954  0.255835  1.033814  
27006  0.124663  0.723003  
27024  0.130570  0.707569  
27072  0.316772  1.410945  

[796 rows x 65 columns]

#### 796 Individuals didn't answer at least one question
That's less than 5 percent of the entire sample. I think that isn't that significant, but it's definintly something we can talk about

### However, there were also individuals who answered with "Don't know"

In [11]:
search98 = skinny_df.apply(pd.value_counts)
search98 = search98.loc['98']
search98.sum()

4392.0

In [12]:
grab98 = skinny_df[(skinny_df == '98').any(axis=1)]
grab98

NFCSID  STATEQ  CENSUSDIV  CENSUSREG  A3  A3Ar_w  A3B  A4A_new_w  \
3      2018010004      10          5          3   2       1    7          1   
5      2018010006      11          5          3   2       2    8          1   
6      2018010007      11          5          3   1       2    2          1   
7      2018010008      26          4          2   2       1    7          1   
17     2018010018      24          4          2   2       5   11          1   
...           ...     ...        ...        ...  ..     ...  ...        ...   
27058  2018037059      51          8          4   2       3    9          1   
27066  2018037067      51          8          4   2       6   12          1   
27067  2018037068      51          8          4   2       3    9          2   
27076  2018037077      51          8          4   2       3    9          1   
27078  2018037079      51          8          4   2       2    8          1   

       A5_2015  A6  ...  M21_3  M21_4  M41  M42  M6  M9 M10    wgt_n2  \
3            6   2  ...      1           3    7  98   2  98  0.808358   
5            5   1  ...                          98  98  98  1.017910   
6            3   1  ...                           1   1   1  1.433500   
7            2   2  ...                           2   1   2  0.445837   
17           2   1  ...                          98   1  98  0.522265   
...        ...  ..  ...    ...    ...  ...  ...  ..  ..  ..       ...   
27058        6   1  ...                          98  99  99  0.320832   
27066        4   1  ...                          98   1   1  0.469334   
27067        6   1  ...                           1   2   2  0.620629   
27076        4   2  ...                           1   2  98  0.404002   
27078        6   2  ...      2           2    4   1   1  98  0.318172   

         wgt_d2    wgt_s3  
3      2.516841  0.922693  
5      1.329637  0.936126  
6      1.479585  1.020436  
7      1.745104  0.931951  
17     1.833441  1.007209  
...         ...       ...  
27058  0.139948  0.378946  
27066  0.201007  1.144149  
27067  0.170350  0.504043  
27076  0.183442  0.808332  
27078  0.146126  0.353300  

[2921 rows x 65 columns]

### 4392 answers of "I don't know" and 2921 individuals with at least one "IDK"
This quite a bit larger than the unanswered questions, with roughly 10.7 percent of the participants answering with "I dont' know" to at least one question